### PR TITLE
Adding log text to warn when Processor.stop-start apis used in resume-pause style

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java [hunk 1]
```diff
@@ -242,6 +243,13 @@ public synchronized void start() {
             return;
         }
 
+        if (wasStopped) {
+            wasStopped = false;
+            LOGGER.warning("Starting Processor that was stopped before is not recommended, and this feature may be deprecated in the future. "
+                + "Please close this processor instance and create a new one to restart processing. "
+                + "Refer the git-ticket https://github.com/Azure/azure-sdk-for-java/issues/34464.");
```
Comment (lmolkova) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java:250]:
```suggestion
                + "Refer to the GitHub issue https://github.com/Azure/azure-sdk-for-java/issues/34464. for more details");
```

?

**Review by lmolkova — APPROVED** [review]
Left a tiny non-blocking suggestion

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java (hunk 1 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java:250]: Liked it, let’s use this.

**Review by conniey — APPROVED** [review]
Thanks!

**Pull request merged**